### DISCUSSION TITLE
TrguiNG - new cross platform client and web interface

### DISCUSSION THREAD

**Discussion by qu1ck** [transmission/transmission#5629] [Show and tell]
I just published first version of TrguiNG

https://github.com/openscopeproject/TrguiNG

Builds for all platforms are available in release page. 

I consider it to be in beta: stable, functional and usable but may need some polish here and there. Please report any issues you may find if you try this app.

Here is main window
![img](https://i.imgur.com/dEJSRyF.png)

There are more screenshots on github wiki page.

**Comment by qu1ck** [comment]
Here are some features that make it stand out from most clients:

* Multi tabbed interface for concurrent server connections (native app only)
* Torrent creation with fast multi threaded hashing (native app only)
* Powerful torrent filtering options
* Latest transmission features support: labels, bandwidth groups, sequential download
* Dark and white theme
* Works as windows/linux/mac app or as a web interface served by transmission itself

**Comment by Pentaphon** [comment]
@qu1ck Looks nice. Trying it out on the latest Firefox!

You may also want advise people on your README that it's even easier to install as a WebUI by using Method 2 as described here: https://github.com/6c65726f79/Transmissionic#method-2 

Suggestions: 

1) You may want to simplify the default columns by disabling the not-commonly-used ones. Not everybody needs to see the bandwidth group of their torrent. I had to delete more than half the columns for my use case. I'd go with as few columns enabled by default as possible. Just the essentials. People can add whatever they want if they want. Here's what I ended up with in the end. I kept it simple like a desktop torrent client does. 

![col](https://github.com/transmission/transmission/assets/49374058/99ceac27-7a6c-477d-a0e5-fc86409642e0)

2) Can you make the server statistics easier to read? Right now they are kind of too spaced out and cause a lot of eye travel on a desktop browser. I'd like a nice small listing to read without by eyes moving all over the place. Maybe align all the info to the left where the tabs are?

3) The general tab is also a bit too spaced out on a desktop browser. Lots of space in between all the useful info. Maybe tighten that up a bit like the server statistics should be?  Maybe align all the info to the left where the tabs are?

4) Could you make the progress bar in the torrent list and the progress bar in the general tab look the same at all times? They are different-looking when downloading a torrent, the torrent list bar has a \\\\\ design when downloading while the general tab progress bar does not. Not sure if its just my eyes, but I don't think the progress bar has the same shade of blue as the progress bar in the general tab does when seeding? 

5) Could you make the "stopped" icon a square like the other icons instead of a rectangle? 

![Screenshot 2023-06-20 at 18-06-06 Screenshot](https://github.com/transmission/transmission/assets/49374058/6fe44d76-e650-46cb-bb7f-c6ddaf32da7d)

6) The X and the clock hands on the error and waiting icons are a bit hard to see on a smaller screen in dark mode. Perhaps make the lines a bit thicker?

![Screenshot 2023-06-20 at 18-04-20 Screenshot](https://github.com/transmission/transmission/assets/49374058/3878ee04-7131-416b-985b-df159909fcc3)

7) Can we also sort by trackers as we can by status, directories and labels?

8) The info in the bottom taskbar is a little too tightly bound together. It needs to be selectively spaced out to be more readable and maybe some of that info is better off elsewhere on the client UI like between the search bar and preferences button? Or next to the statistics button? Not sure, but it's a bit hard to read all that info on the status bar when there's barely any spacing or separators.  Not sure if the total space or selected space needs to be shown, either. I think the most important is showing the free space. 

9) The strong blue color when selecting a torrent on the torrent list seems to overpower the lighter, cyanish-y blue color of the progress bar in both light and dark modes and seems to especially overpower the blue downloading icon when downloading a torrent in light mode. I suggest going with a much lighter blue color or maybe some other color for the selection color for each mode. 

![selection](https://github.com/transmission/transmission/assets/49374058/07c9d652-ce54-475f-8d6e-cab765f660e4)
![selectdark](https://github.com/transmission/transmission/assets/49374058/7aba8ae1-047d-4c59-b739-646aacff745b)

10) When you add a torrent, the current download directory field isn't already pre-filled like it is on the official WebUI.

![dl](https://github.com/transmission/transmission/assets/49374058/bdd4ca79-946e-45e0-bf43-2a51144ceb99)

11) The "active" icon is much less "icon-y" and harder to make out than the rest of the icons. Perhaps make it the same as TransGUI, the up and down arrows inside an orange square? But following the style of the other square icons? This would be much more visible than the current "pulse" icon. 

![active](https://github.com/transmission/transmission/assets/49374058/ee8a9154-4367-495e-8537-50003fdae175)

**Reply by qu1ck** [reply]
Thanks for the feedback. Let me reply on each point

1, Minimal set of useful fields is a very subjective thing, different people use different columns. I don't think trying to keep it compact by default is useful, there is no good one size fits all solution here and first time setup is expected. That said, some fields are actually very rarely used and I may make them disabled by default.

2, 3 Agreed, I'll look into this

4, progress bar in table has animated barber pole (\\\ design) when there is activity on the torrent (i.e. upload or download speed > 0). It should be same shade of blue, I'll double check.

5, 6, 11, good points, I don't really use custom icons right now, all of the icons are from bootstrap-icons project and those icons were the best fit I could find. I'll have to draw custom ones later, I'm focusing on UX and functionality first. 

7, You mean filter by trackers? Already added, will be in v0.3.0 (probably released today)

8, There is more spacing in v0.2.0 compared to v0.1.0, which version did you use? Which thing is most important is again subjective, I look at speeds and torrent sizes all the time. I think with added spacing in v0.2.0 information is easy to digest. But it's something that is easy to make configurable like filter sections, I'll add that.

9, good points

10, it will prefill last used folder on next torrent addition and remember a number of last used folders that you have access to from the clock icon in the right section of the download directory textbox.

**Reply by Pentaphon** [reply]
>7, You mean filter by trackers? Already added, will be in v0.3.0 (probably released today)

Yes, that's what I meant. Very nice!

>8, There is more spacing in v0.2.0 compared to v0.1.0, which version did you use? 

Sorry, I should have mentioned that. I used 0.2.0 on Firefox on Windows 11. 

>10, it will prefill last used folder on next torrent addition and remember a number of last used folders that you have access to from the clock icon in the right section of the download directory textbox.

Ah I see thanks for clearing that up.
